### DISCUSSION TITLE
性能优化提醒

### DISCUSSION THREAD

**Discussion by peter100u** [Tencent/libpag#2375] [Ideas] [answered]
首先感谢libpag的持续产出，让我们有了很多的特效选择，由于产品版本一直在迭代，libpag也一直在升级， 后期出现了很多性能问题，但是很难定位，特别问题出现在资源本身上，我知道pag提供了pagviewer的播放器可以看到各项指标， 然后特效组的产出是和运营同学直接对接的，并不会在线上阶段同步产品同学的，所以希望libpag在日志中把资源的性能数据打印出来，提供给开发者进行观察和日志分析，线上大量的资源，不可能通过人工去一个一个查看，只能通过日志

**Comment by shixiaoda** [comment]
很好的建议

**Comment by domchen** [comment, marked as answer]
线上排查其实也已经比较晚了，那时候已经造成用户体验的影响了。其实你说的这个场景我们有更好的实践方案，我们在SDK是提供了检查性能的接口的。你可以自己编译一下PAG的Linux版本，只需要静态库就行。然后写个很简单的命令行程序去引用 pag/file.h 这个头文件，使用 Codec::ReadPerformanceData()接口就可以读取每个PAG文件的性能数据。这个性能数据是要设计师在PAGViewer菜单里点一下「文件-性能校验」保存到文件的。如果设计师没有点击这个的话，你读取不到数据就认为这个文件非法。读取到数据之后你设置一个阈值来判断性能。这样做成一个服务，就可以集成到后台资源上传检测的过程里，一劳永逸。
